File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.py
from collections import Counter

def solution(topping):
    answer = 0
    brother = Counter(topping)
    brother_kind = len(brother)
    
    chulsoo = set()
    
    for t in topping:
        chulsoo.add(t)
        
        brother[t] -= 1
        if brother[t] == 0:
            brother_kind -= 1
            
        if len(chulsoo) == brother_kind:
            answer += 1
            
    return answer

File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.js
function solution(topping) {
    var answer = 0;
    
    let brother = new Map();
    for (const item of topping) {
        brother.set(item, (brother.get(item) || 0) + 1);
    }
    let brotherCount = brother.size;
    
    let chulsoo = new Map();
    
    for (let t of topping) {
        chulsoo.set(t, (chulsoo.get(t) || 0) + 1);
        
        brother.set(t, (brother.get(t) || 0) - 1);
        if (brother.get(t) === 0) {
            brotherCount -= 1;
        }
        
        if (chulsoo.size === brotherCount) {
            answer++;
        }
    }
    
    return answer;
}